processing/src/main/java/org/apache/druid/segment/IndexMergerV9.java

```diff
@@ -1460,4 +1456,33 @@ private static <T extends TimeAndDimsPointer> T reorderRowPointerColumns(
       );
     }
   }
+
+  private static class DimensionMergerUtil
```

Comment:
super nit, this doesn't really seem like a 'util', but does sort of remind me of some other thingswe call `Inspector` though not sure what exactly would call this, naming is hard.

Also, it seems like only one place creates this, maybe should push `dimensionsSpec` into the constructor of this thing and do the logic there.

Reply:
+1

In that vein, maybe `DimensionsSpecInspector` or `DimensionStorageChecker` would be a better name?